pkg/apk/implementation.go

```diff
@@ -988,6 +997,8 @@ func packageAsURI(pkg InstallablePackage) (uri.URI, error) {
 
 	if strings.HasPrefix(u, "https://") {
 		return uri.Parse(u)
+	} else if strings.HasPrefix(u, "s3://") {
+		return uri.URI(u), nil
```

Comment:
It's worth calling out that this line has a bit of code-smell, `uri.Parse` will fail for the `s3` schema: https://go.dev/play/p/PlTtp5oC7S3

This line currently just directly casts to the underlying `uri.URI` type to avoid this which isn't great